core/src/main/java/org/mskcc/cbio/oncokb/util/ClinicalTrialsUtils.java

```diff
@@ -0,0 +1,199 @@
+
+            if (treatment == null) {
+                return new ArrayList<Trial>(tumor.getTrials());
+            }
+
+            List<Trial> trial = getTrialByTreatment(tumor.getTrials(), treatment);
+            return new ArrayList<Trial>(trial);
+        }
+        return new ArrayList<Trial>(new ArrayList<>());
+    }
+
+    public Map<String, List<Trial>> trialsGetByCancerTypes(CancerTypesQuery body)
+        throws UnsupportedEncodingException, IOException, ParseException {
+        Map<String, List<Trial>> result = new HashMap<>();
+
+            JSONObject trialsJSON = CacheUtils.getTrialsJSON();
+            JSONObject oncotreeMappingJSON = CacheUtils.getOncoTreeMappingTrials();
+
+            Set<String> cancerTypes = new HashSet<>(body.getCancerTypes());
+            if (cancerTypes.contains(SpecialTumorType.ALL_TUMORS.getTumorType())) {
+                List<Trial> trials = new ArrayList<>();
+                Set<String> nctIDSet = new HashSet<>();
+                for (Object item : oncotreeMappingJSON.keySet()) {
+                    String oncoTreeCode = (String) item;
+                    Tumor tumor = getTumor(oncotreeMappingJSON, trialsJSON, oncoTreeCode);
+
+                    for (Trial curTrial : tumor.getTrials()) {
+                        if (!nctIDSet.contains(curTrial.getNctId())) {
+                            nctIDSet.add(curTrial.getNctId());
+                            trials.add(curTrial);
+                        }
+                    }
+                }
+                result.put(SpecialTumorType.ALL_TUMORS.getTumorType(), trials);
+                return result;
+            }
+
+            for (String cancerType : cancerTypes) {
+                Set<String> nctIDSet = new HashSet<>();
+                List<Trial> addTrials = new ArrayList<>();
+                List<Trial> trials = new ArrayList<>();
+                SpecialTumorType specialTumorType = ApplicationContextSingleton.getTumorTypeBo().getSpecialTumorTypeByName(cancerType);
+                if (specialTumorType != null) {
+                    trials = getTrialsForSpecialCancerType(oncotreeMappingJSON, trialsJSON, specialTumorType);
+                } else {
+                    trials = getTrialsByCancerType(oncotreeMappingJSON, trialsJSON, cancerType);
+                }
+                for (Trial trial : trials) {
+                    if (!nctIDSet.contains(trial.getNctId())) {
+                        nctIDSet.add(trial.getNctId());
+                        addTrials.add(trial);
+                    }
+                }
+                result.put(cancerType, addTrials);
+        }
+        return result;
+    }
+
+    private Tumor getTumor(JSONObject oncotreeMappingJSON, JSONObject trialsJSON, String oncoTreeCode) {
+        Tumor tumor = new Tumor();
+        if (oncotreeMappingJSON.containsKey(oncoTreeCode)) {
+            JSONObject tumorObj = (JSONObject) oncotreeMappingJSON.get(oncoTreeCode);
```

Comment:
Same goes here, we could just convert oncotreeMappingJSON to a Map<String, Tumor>